pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -31,10 +31,16 @@ import (
 type Code int
 
 // NodeScoreList declares a list of nodes and their scores.
-type NodeScoreList []int
+type NodeScoreList []NodeScore
```

Comment:
Wouldn't it be better to use a map from node names to node score (`map[string]int`) instead of `NodeScoreList`? With the current data structure, finding the score of a node requires iterating over a NodeScoreList.

Reply:
> Wouldn't it be better to use a map from node names to node score (`map[string]int`) instead of `NodeScoreList`? With the current data structure, finding the score of a node requires iterating over a NodeScoreList.

I don't have a preference here

> I recalled that in the [API convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps) documents, kubernetes prefer the named array to map. I could go with the second option if it is the convention we must follow here, but I prefer the first for now. What do you think of this?

Reply:
We have to think carefully about this question: "Do we need to find score of a particular node in our code?" If the answer is no, we should definitely go with the current List instead of a map. Lists are faster to access compared to maps if we don't need to search for something.

Reply:
> We have to think carefully about this question: "Do we need to find score of a particular node in our code?"

+1

> If the answer is no, we should definitely go with the current List instead of a map. Lists are faster to access compared to maps if we don't need to search for something.

For the implementation of InterPodAffinity and EvenPodSpread, we don't need to find the score of a particular node in NormalizeScore, and we use node name in the named array (or map) to find score from the internal data structure.

Reply:
Unless we have a strong reason to use map (which I doubt we will), my recommendation is to use a list.